### PR TITLE
Replace zu with PRIsz and llu with PRIu64.

### PR DESCRIPTION
**Pull request by SeanCampbell**
The new macro, PRIsz, can be defined differently on systems that don't support PRIu64, zu, and llu, which will make the code easier to port.

### CODE REVIEW

**Comment by SeanCampbell** [comment]
@igorpeshansky - Can you take a look at this? This is based on the [discussion](https://github.com/Stackdriver/collectd/commit/bee5e168175e3078b681585fa0fe3f7361d90055) in the windows-fixes branch in the Stackdriver fork.

File: src/daemon/common.h [hunk 1]
```diff
@@ -384,4 +384,8 @@ void strarray_free(char **array, size_t array_len);
 int check_capability(int arg);
 #endif /* HAVE_SYS_CAPABILITY_H */
 
+#ifndef PRIsz
+#define PRIsz PRIu64
```
Comment (igorpeshansky) [inline src/daemon/common.h]: Is there a reason you didn't use the [latest version](https://github.com/Stackdriver/collectd/blob/windows-fixes/src/daemon/common.h#L390) of those changes?

File: src/utils_format_graphite.c [hunk 2]
```diff
@@ -60,11 +60,11 @@ static int gr_format_values(char *ret, size_t ret_len, int ds_num,
   else if (rates != NULL)
     BUFFER_ADD("%f", rates[ds_num]);
   else if (ds->ds[ds_num].type == DS_TYPE_COUNTER)
-    BUFFER_ADD("%llu", vl->values[ds_num].counter);
+    BUFFER_ADD("%" PRIsz, (uint64_t)vl->values[ds_num].counter);
```
Comment (igorpeshansky) [inline src/utils_format_graphite.c]: I think you need to use `PRIu64` for `uint64_t` throughout.

File: src/utils_format_graphite.c [hunk 3]
```diff
@@ -60,11 +60,11 @@ static int gr_format_values(char *ret, size_t ret_len, int ds_num,
...
   else if (ds->ds[ds_num].type == DS_TYPE_DERIVE)
     BUFFER_ADD("%" PRIi64, vl->values[ds_num].derive);
   else if (ds->ds[ds_num].type == DS_TYPE_ABSOLUTE)
-    BUFFER_ADD("%" PRIu64, vl->values[ds_num].absolute);
+    BUFFER_ADD("%" PRIsz, vl->values[ds_num].absolute);
```
Comment (igorpeshansky) [inline src/utils_format_graphite.c]: Ditto here, you need `PRIu64` for `uint64_t`.

File: src/daemon/common.h (hunk 1 above)
Comment (SeanCampbell) [reply src/daemon/common.h]: I left out the Windows bits, since it doesn't currently run on Windows, so I couldn't make sure it all worked.

If you want, I can add them in, since it is just copying and pasting over.

File: src/utils_format_graphite.c (hunk 2 above)
Comment (SeanCampbell) [reply src/utils_format_graphite.c]: Fixed

File: src/utils_format_graphite.c (hunk 3 above)
Comment (SeanCampbell) [reply src/utils_format_graphite.c]: Thanks, I didn't quite get where to use PRIu64 before. Fixed.

File: src/daemon/common.h (hunk 1 above)
Comment (igorpeshansky) [reply src/daemon/common.h]: Nah, I was proposing to define `PRIsz` in terms of what works on Linux for now, so you could add additional cases later.

**Comment by igorpeshansky** [comment]
@octo @tokkee This should now be ready for review...

File: src/postgresql.c [hunk 4]
```diff
@@ -742,7 +742,7 @@ static char *values_to_sqlarray(const data_set_t *ds, const value_list_t *vl,
 
       status = snprintf(str_ptr, str_len, ",%lf", rates[i]);
     } else if (ds->ds[i].type == DS_TYPE_COUNTER)
-      status = snprintf(str_ptr, str_len, ",%llu", vl->values[i].counter);
+      status = snprintf(str_ptr, str_len, ",%" PRIu64, vl->values[i].counter);
```
Comment (igorpeshansky) [inline src/postgresql.c]: Do you need a cast here?

File: src/rrdcached.c [hunk 5]
```diff
@@ -88,7 +88,7 @@ static int value_list_to_string(char *buffer, int buffer_len,
       return -1;
 
     if (ds->ds[i].type == DS_TYPE_COUNTER) {
-      status = snprintf(buffer + offset, buffer_len - offset, ":%llu",
+      status = snprintf(buffer + offset, buffer_len - offset, ":%" PRIu64,
                         vl->values[i].counter);
```
Comment (igorpeshansky) [inline src/rrdcached.c]: Do you need a cast here?

File: src/rrdtool.c [hunk 6]
```diff
@@ -226,7 +226,7 @@ static int value_list_to_string(char *buffer, int buffer_len,
                       vl->values[0].gauge);
     break;
   case DS_TYPE_COUNTER:
-    status = snprintf(buffer, buffer_len, "%u:%llu", (unsigned)tt,
+    status = snprintf(buffer, buffer_len, "%u:%" PRIu64, (unsigned)tt,
```
Comment (igorpeshansky) [inline src/rrdtool.c]: Do you need a different cast here?

File: src/snmp.c [hunk 7]
```diff
@@ -1093,7 +1093,7 @@ static int csnmp_instance_list_add(csnmp_list_instances_t **head,
     value_t val = csnmp_value_list_to_value(
         vb, DS_TYPE_COUNTER,
         /* scale = */ 1.0, /* shift = */ 0.0, hd->name, dd->name);
-    snprintf(il->instance, sizeof(il->instance), "%llu", val.counter);
+    snprintf(il->instance, sizeof(il->instance), "%" PRIu64, val.counter);
```
Comment (igorpeshansky) [inline src/snmp.c]: Do you need a cast here?

File: src/utils_format_kairosdb.c [hunk 8]
```diff
@@ -154,7 +154,7 @@ static int values_to_kairosdb(char *buffer, size_t buffer_size, /* {{{ */
     BUFFER_ADD("[[");
     BUFFER_ADD("%" PRIu64, CDTIME_T_TO_MS(vl->time));
     BUFFER_ADD(",");
-    BUFFER_ADD("%llu", vl->values[ds_idx].counter);
+    BUFFER_ADD("%" PRIu64, vl->values[ds_idx].counter);
```
Comment (igorpeshansky) [inline src/utils_format_kairosdb.c]: Do you need a cast here?

File: src/virt.c [hunk 9]
```diff
@@ -699,8 +699,9 @@ static double cpu_ns_to_percent(unsigned int node_cpus,
               (time_diff_sec * node_cpus * NANOSEC_IN_SEC);
   }
 
-  DEBUG(PLUGIN_NAME ": node_cpus=%u cpu_time_old=%llu cpu_time_new=%llu"
-                    "cpu_time_diff=%llu time_diff_sec=%f percent=%f",
+  DEBUG(PLUGIN_NAME ": node_cpus=%u cpu_time_old=%" PRIu64
+                    " cpu_time_new=%" PRIu64 "cpu_time_diff=%" PRIu64
+                    " time_diff_sec=%f percent=%f",
         node_cpus, cpu_time_old, cpu_time_new, cpu_time_diff, time_diff_sec,
```
Comment (igorpeshansky) [inline src/virt.c]: Do you need casts here?

File: src/curl_xml.c [hunk 10]
```diff
@@ -240,8 +240,9 @@ static int cx_check_type(const data_set_t *ds, cx_xpath_t *xpath) /* {{{ */
   }
 
   if (ds->ds_num != xpath->values_len) {
-    WARNING("curl_xml plugin: DataSet `%s' requires %zu values, but config "
-            "talks about %zu",
+    WARNING("curl_xml plugin: DataSet `%s' requires %" PRIsz
+            " values, but config "
+            "talks about %" PRIsz,
```
Comment (igorpeshansky) [inline src/curl_xml.c]: Join with the previous line.

File: src/email.c [hunk 11]
```diff
@@ -373,14 +373,13 @@ static void *open_connection(void __attribute__((unused)) * arg) {
     pthread_exit((void *)1);
   }
 
-  struct sockaddr_un addr = {
-    .sun_family = AF_UNIX
-  };
+  struct sockaddr_un addr = {.sun_family = AF_UNIX};
```
Comment (igorpeshansky) [inline src/email.c]: I would undo this change -- it reads better as multiple lines. But up to @octo.

File: src/email.c [hunk 12]
```diff
@@ -373,14 +373,13 @@ static void *open_connection(void __attribute__((unused)) * arg) {
...
   sstrncpy(addr.sun_path, path, (size_t)(UNIX_PATH_MAX - 1));
 
   errno = 0;
   if (bind(connector_socket, (struct sockaddr *)&addr,
-           offsetof(struct sockaddr_un, sun_path) + strlen(addr.sun_path)) == -1) {
+           offsetof(struct sockaddr_un, sun_path) + strlen(addr.sun_path)) ==
+      -1) {
```
Comment (igorpeshansky) [inline src/email.c]: I'd undo this change. But up to @octo.

File: src/libcollectdclient/network_parse_test.c [hunk 13]
```diff
@@ -245,10 +245,9 @@ static int test_network_parse() {
     uint8_t buffer[LCC_NETWORK_BUFFER_SIZE_DEFAULT];
     size_t buffer_size = sizeof(buffer);
     if (decode_string(raw_packet_data[i], buffer, &buffer_size)) {
-      fprintf(
-          stderr,
-          "lcc_network_parse(raw_packet_data[%zu]): decoding string failed\n",
-          i);
+      fprintf(stderr, "lcc_network_parse(raw_packet_data[%" PRIsz
```
Comment (igorpeshansky) [inline src/libcollectdclient/network_parse_test.c]: How about:
```c
      fprintf(stderr, "lcc_network_parse(raw_packet_data[%" PRIsz "]):"
                      " decoding string failed\n",
              i);
```
?

File: src/nfs.c [hunk 14]
```diff
@@ -488,7 +488,7 @@ static int nfs_submit_nfs4_client(const char *instance, char **fields,
     if (!suppress_warning) {
       WARNING("nfs plugin: Unexpected number of "
               "fields for NFSv4 %s "
-              "statistics: %zu. ",
+              "statistics: %" PRIsz ". ",
```
Comment (igorpeshansky) [inline src/nfs.c]: Join with the previous line?

File: src/oracle.c [hunk 15]
```diff
@@ -532,7 +532,8 @@ static int o_read_database_query(o_database_t *db, /* {{{ */
     memcpy(column_names[i], column_name, column_name_length);
     column_names[i][column_name_length] = 0;
 
-    DEBUG("oracle plugin: o_read_database_query: column_names[%zu] = %s; "
+    DEBUG("oracle plugin: o_read_database_query: column_names[%" PRIsz
```
Comment (igorpeshansky) [inline src/oracle.c]: How about:
```c
    DEBUG("oracle plugin: o_read_database_query: column_names[%" PRIsz "] = %s;"
          " column_name_length = %" PRIu32 ";",
          i, column_names[i], (uint32_t)column_name_length);
```
?

File: src/utils_lua.c [hunk 16]
```diff
@@ -57,8 +57,9 @@ static int ltoc_values(lua_State *L, /* {{{ */
   } /* while (lua_next) */
 
   if (i != ds->ds_num) {
-    WARNING("ltoc_values: invalid size for datasource \"%s\": expected %zu, "
-            "got %zu",
+    WARNING("ltoc_values: invalid size for datasource \"%s\": expected %" PRIsz
+            ", "
+            "got %" PRIsz,
```
Comment (igorpeshansky) [inline src/utils_lua.c]: Join with the previous line?

**Comment by igorpeshansky** [comment]
@SeanCampbell, you might also want to update the PR description (e.g., to `Replace "llu" with PRIu64 and "zu" with PRIsz in format strings`).
@octo, @tokkee, once this is approved, it would probably make sense to either squash-and-merge or rewrite the history on the PR branch before merging.

File: src/curl_xml.c (hunk 10 above)
Comment (SeanCampbell) [reply src/curl_xml.c]: Good call. clang-format formatted it this way, but it looks like it won't combine adjacent strings, only split long ones.

File: src/email.c (hunk 11 above)
Comment (SeanCampbell) [reply src/email.c]: The last commit is just changes from running clang-format, as per the contributions guidelines [1], so I'm guessing this is correct but could be missing something.

[1] https://github.com/collectd/collectd/blob/66fd3303352c526255ab4bde256361244e26b259/CONTRIBUTING.md

File: src/postgresql.c (hunk 4 above)
Comment (SeanCampbell) [reply src/postgresql.c]: I think you're right with all of these. I was able to build and install without error, but it probably wasn't building these plugins and so didn't throw any error.

**Comment by SeanCampbell** [comment]
Friendly ping :)

Should I deal with the conflicting files, or is that dealt with as the PR is merged?

**Comment by SeanCampbell** [comment]
@octo @tokkee Friendly ping on this :) Is there anything I should clean up before this is reviewed or merged?

File: src/email.c (hunk 11 above)
Comment (tokkee) [reply src/email.c]: Yeah, this is clang-format, so that's fine.

**Comment by igorpeshansky** [comment]
@tokkee A general comment: GitHub now has squash-and-merge. Unless we want to merge this as multiple commits for some reason, wouldn't the final squash-and-merge "just work"?

**Comment by SeanCampbell** [comment]
Merged the conflicts.

For rewriting the branch, do you mean just combining the commits for this PR into one? Thanks.

**Comment by SeanCampbell** [comment]
@tokkee Cleaned up the history commit -- what do you think?

**Comment by SeanCampbell** [comment]
Friendly ping

**Comment by tokkee** [comment]
Thanks! I would have been fine with using squash-and-merge but wanted to leave it to Sean to decide how to split this into one or more commits.
